internal/services/datafactory/data_factory_linked_service_sql_server_resource.go

```diff
@@ -266,6 +275,7 @@ func resourceDataFactoryLinkedServiceSQLServerRead(d *pluginsdk.ResourceData, me
 	}
 
 	if properties := sqlServer.SQLServerLinkedServiceTypeProperties; properties != nil {
+		d.Set("user_name", properties.UserName)
```

Comment:
should this be nil checked before setting?

Reply:
Tanks for this comment @catriona-m，`nil` will be checked before setting in the `Set` function